io-sim/src/Control/Monad/IOSim/Internal.hs

```diff
@@ -119,7 +119,7 @@ labelledThreads threadMap =
 data TimerCompletionInfo s =
        Timer !(TVar s TimeoutState)
      | TimerRegisterDelay !(TVar s Bool)
-     | TimerThreadDelay !ThreadId
+     | TimerThreadDelay !ThreadId !TimeoutId
```

Comment:
This is only for debugging purposes? I am happy to leave it here but maybe a small comment just to help remember what that's for!

Reply:
We use `TimeoutId` in `TimerTimeout` probably for exactly the same reason.

Reply:
Done.